src/main/java/net/neoforged/neoforge/server/command/TagsCommand.java

```diff
@@ -171,16 +175,45 @@ private static MutableComponent createMessage(final MutableComponent header,
             final long currentPage,
             final ChatFormatting elementColor,
             final Supplier<Stream<String>> names) {
-        final String allElementNames = names.get().sorted().collect(Collectors.joining("\n"));
         final long totalPages = (count - 1) / PAGE_SIZE + 1;
         final long actualPage = (long) Mth.clamp(currentPage, 1, totalPages);
 
         MutableComponent containsComponent = Component.translatable(containsText, count);
         if (count > 0) // Highlight the count text, make it clickable, and append page counters
         {
+            final String clipboardText;
+            final StringBuilder clipboardTextBuilder = new StringBuilder();
+            boolean reachedLimit = false;
+            int countedLines = 0;
+
+            Iterator<String> iterator = names.get().sorted().iterator();
+            while (iterator.hasNext()) {
+                final String line = iterator.next();
+                if (clipboardTextBuilder.length() + line.length() > CLIPBOARD_TEXT_LIMIT) {
+                    // The to-be-added line puts us over the limit, so stop adding lines
+                    reachedLimit = true;
+                    break;
+                }
+                clipboardTextBuilder.append(line).append('\n');
+                countedLines++;
+            }
+            // Remove the trailing newline if present
+            if (!clipboardTextBuilder.isEmpty()) {
+                clipboardTextBuilder.deleteCharAt(clipboardTextBuilder.length() - 1);
+            }
+
+            if (reachedLimit) {
+                // Almost went over the limit; add additional info to clipboard text
+                clipboardText = "(Too many entries to fit in clipboard, showing only first " + countedLines + " entries...)" + '\n'
```

Comment:
Shouldn't there be a `\n` at the beginning since you remove it just above?

Reply:
What do you mean? The clipboard text builder will contain a string that doesn't end in a newline, and in the case of the limit being reached, the starting truncation notice and the ending truncation notice are both separated by a newline from the main (elements) text.

Reply:
Ah right this looks correct.